library/std/src/sys/unix/thread.rs

```diff
@@ -265,12 +265,21 @@ impl Drop for Thread {
 
 pub fn available_concurrency() -> io::Result<NonZeroUsize> {
     cfg_if::cfg_if! {
-        if #[cfg(any(
```

Comment:
Instead of duplicating the `match unsafe { libc::sysconf(libc::_SC_NPROCESSORS_ONLN) }` part, it might be clearer to conditionally add the new code to the existing branch, keeping a single sysconf(_SC_NPROCESSORS_ONLN) call:

```rust
    cfg_if::cfg_if! {
        if #[cfg(any(
            target_os = "android",
            target_os = "emscripten",
            target_os = "fuchsia",
            target_os = "ios",
            target_os = "linux",
            target_os = "macos",
            target_os = "solaris",
            target_os = "illumos",
        ))] {
            #[cfg(any(target_os = "android", target_os = "linux"))]
            {
                let mut set: libc::cpu_set_t = unsafe { mem::zeroed() };
                if unsafe { libc::sched_getaffinity(0, mem::size_of::<libc::cpu_set_t>(), &mut set) } == 0 {
                    let count = unsafe { libc::CPU_COUNT(&set) };
                    return Ok(unsafe { NonZeroUsize::new_unchecked(count as usize) });
                }
            }
            match unsafe { libc::sysconf(libc::_SC_NPROCESSORS_ONLN) } {
                -1 => Err(io::Error::last_os_error()),
                0 => Err(io::Error::new_const(io::ErrorKind::NotFound, &"The number of hardware threads is not known for the target platform")),
                cpus => Ok(unsafe { NonZeroUsize::new_unchecked(cpus as usize) }),
            }
    } ...
```

Reply:
Done. I think the use of `cfg_if` had made me momentarily forget that std (unlike `libc`) doesn't *require* the use of `cfg_if`.